src/util/templates/TemplateEngineHandler.ts

```diff
@@ -0,0 +1,55 @@
+import { NotImplementedHttpError } from '../errors/NotImplementedHttpError';
+import { AsyncHandler } from '../handlers/AsyncHandler';
+import { getExtension } from '../PathUtil';
+import type { Template, TemplateEngine, TemplateEngineInput } from './TemplateEngine';
+import { getTemplateFilePath } from './TemplateEngine';
+import Dict = NodeJS.Dict;
+
+/**
+ * Parent class for TemplateEngine implementations.
+ * This class enables TemplateEngine implementations to be used as AsyncHandlers.
+ */
+export abstract class TemplateEngineHandler<T extends Dict<any> = Dict<any>>
+  extends AsyncHandler<TemplateEngineInput<T>, string> implements TemplateEngine<T> {
```

Comment:
The idea behind the issue was that we no longer would have the interface `TemplateEngine` but would simply have an `abstract class TemplateEngine extends AsyncHandler<...>`. The current solution is somewhere inbetween and keeps too much of the old solution in my opinion. The `render` function should just disappear and be replaced by the usual `handle`/`handleSafe` calls where it is needed.

A `TemplateUtil.ts`  file can be added for all the utility now in `TemplateEngine`. If some relevant behaviour is shared between the template engines that fits better in a class than utility functions, we can have an abstract class `BaseTemplateEngine` that extends `TemplateEngine`. This would bring all of this more in line with how all other class hierarchies are implemented in the server. Might even need a secondary `ExtensionBasedTemplateEngine` that extends this `BaseTemplateEngine` as it feels a bit weird now that `GenericTemplateEngine` also extends a class with that feature.